pkg/archive/copy.go

```diff
@@ -103,22 +106,34 @@ func TarResourceRebase(sourcePath, rebaseName string) (content io.ReadCloser, er
 		return
 	}
 
+	sourceDir, opts := TarResourceRebaseOpts(sourcePath, rebaseName, pathdriver.LocalPathDriver)
+
+	logrus.Debugf("copying %v from %q", opts.IncludeFiles, sourceDir)
+
+	return TarWithOptions(sourceDir, opts)
+}
+
+// TarResourceRebaseOpts does not preform the Tar, but instead just creates the parameters
+// to be sent to TarWithOptions (the source directory and the TarOptions struct)
+func TarResourceRebaseOpts(sourcePath string, rebaseName string, driver pathdriver.PathDriver) (string, *TarOptions) {
```

Comment:
Can we have this function just take in the `sourceDir` and `sourceBase`? That would eliminate the need to pass in the `pathdriver` here and changing the `SplitPathDirEntry`. I would like to eliminate all use of the `pathdriver` in the archive package if possible.

Reply:
Yeah it's possible to do that and have a helper function elsewhere that does the split with the `pathdriver` interface.

Reply:
The last reference to the `pathdriver` is `archive.PreserveTrailingDotOrSeparator`. That's used by `container/archive.go` and `daemon/archive.go`. It's always paired with `system.CheckSystemDriveAndRemoveDriveLetter`, so maybe the container struct can provide a method that does those two like `CleanContainerPath` and then we can remove the `pathdriver` reference from those two functions.